api/src/test/java/org/apache/iceberg/expressions/TestEvaluator.java

```diff
@@ -220,6 +222,24 @@ public void testNotEqual() {
 
   }
 
+  @Test
+  public void testStartsWith() {
+    StructType struct = StructType.of(required(24, "s", Types.StringType.get()));
+    Evaluator evaluator = new Evaluator(struct, startsWith("s", "abc"));
+    Assert.assertTrue("abc startsWith abc => true", evaluator.eval(TestHelpers.Row.of("abc")));
+    Assert.assertFalse("abc startsWith Abc => false", evaluator.eval(TestHelpers.Row.of("Abc")));
+    Assert.assertFalse("a startsWith abc => false", evaluator.eval(TestHelpers.Row.of("a")));
```

Comment:
Can you also add a tests for `abcd startsWith abc`?

Reply:
I can add that. With respect to the syntax, I copied a similar variant to what is found elsewhere. For converting it to typical "should" test syntax, would `abcd startsWith abc should be true` work?

The original syntax for this which I modified somewhat comes from a test in `TestStartsWith` (and possibly elsewhere), which reads as `"startsWith(abcde, truncate(abcdg,2))  => true"` (where the test row is "abcdg"). For me, I have to think about what exactly that statement is expressing a little harder than I'd like, but I will admit I haven't had all of my morning coffee so maybe it's plenty clear expressed that way?

I'm open to suggestions, but I'll switch to `abcd startsWith abc should be true` for now.